Fix Windows Node errors on loading pyodide

robert-w-gries opened the pull request:
<!-- Thank you for contributing to Pyodide! All improvements are welcome,
     so don't be afraid to make a PR. -->

### Description

<!-- Please explain what your PR is about:
     - reasoning for the change
     - some details of updated code
     - any noteworthy choices to be aware of
	Please refer to any related issues by #<issue_id> -->

Fixes #2882 

There were two separate issues:

1. There's a difference in Node's handling of imports across OSes. You can find more info + the recommended solution in this issue: https://github.com/nodejs/node/issues/31710
2. `calculateIndexURL()` did not handle the case where the filename had backslashes on Windows instead of forward slashes. Since the `fileName.lastIndexOf("/")` evaluated to `-1`, the calculated indexURL value was `${/path/to/pyodide}/pyodide.j`

I recommend adding error handling to the `calculateIndexURL()` function to throw when no forward slashes or backslashes are found. I can add that as part of this PR or I can defer if y'all don't want me to add it.

### Checklists

<!-- Note:
     If you think some of these steps are not necessary for your PR,
     remove those checkboxes, or mark them as checked. If you keep unchecked checkboxes,
     we will assume that your PR is not ready to be merged  -->

- [ ] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

ryanking13 commented:
Thanks, @robert-w-gries. Generally LGTM. Could you please update the changelog?

> recommend adding error handling to the `calculateIndexURL()` function to throw when no forward slashes or backslashes are found. I can add that as part of this PR or I can defer if y'all don't want me to add it.

Yes, that sounds reasonable to me. Feel free to add it this PR.

robert-w-gries commented:
Thanks @ryanking13, I've updated the changelog and added the error handling for missing trailing slash. Can you take a look and let me know if any other changes are needed?

ryanking13 commented:
Thanks a lot @robert-w-gries!
